libs/blocks/iframe/iframe.js

```diff
@@ -1,13 +1,57 @@
 import { createTag } from '../../utils/utils.js';
 
+const ALLOWED_MESSAGE_ORIGINS = [
+  'https://stage.plan.adobe.com',
+  'https://plan.adobe.com',
+];
+
+function handleManagePlanEvents(message) {
+  const { subType, data } = message;
+  switch (subType) {
+    case 'EXTERNAL':
+      if (!data?.externalUrl || !data?.target) return;
+      window.open(data.externalUrl, data.target);
+      break;
+    case 'SWITCH':
+      if (!data?.externalUrl || !data?.target) return;
+      window.open(data.externalUrl, data.target);
+      break;
+    case 'Close':
+      document.querySelector('.dialog-modal')?.dispatchEvent(new Event('iframe:modal:closed'));
+      break;
+    default:
+      break;
+  }
+}
+
+export function handleIFrameEvents({ data }) {
+  let parsedMsg = null;
+  try {
+    parsedMsg = JSON.parse(data);
+  } catch (error) {
+    return;
+  }
+  const { app } = parsedMsg;
+  if (app === 'ManagePlan') handleManagePlanEvents(parsedMsg);
+}
+
 export default function init(el) {
-  const url = el.href ?? el.querySelector('a')?.href;
+  const linkHref = el.href ?? el.querySelector('a')?.href;
   el.classList.remove('iframe');
   const classes = [...el.classList].join(' ');
 
-  if (!url) return;
+  let url = null;
+  try {
+    url = new URL(linkHref);
+  } catch (e) {
+    return;
+  }
```

Comment:
According to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#href the href should always be a valid url, so I think we can just do:
```suggestion
  if(!linkHref) return
  let url = new URL(linkHref)
```